rosetta-2021.html

```diff
@@ -0,0 +1,236 @@
+<script>
+  // Rule of thumb: put a semicolon at the end of every line that doesn't end in a curly
+
+  // Simple output
+  // =============
+
+  // puts "howdy!"
+
+  // alert("howdy!");
+
+
+  // Declaring local variables
+  // =========================
+
+  // name = "Raghu"
+  // puts "howdy, #{name}!"
+
+  // var name = "Raghu";
+  // alert("howdy, " + name + "!");
+
+  // Note: there's no String interpolation in JS
+
+
+  // Defining and calling methods
+  // ============================
+
+  // def greet
+  //   puts "howdy!"
+  // end
+
+  // greet
+
+  // function greet() {
+  //   alert("howdy!");
+  // }
+
+  // greet();
+
+
+  // Methods that accept arguments
+  // =============================
+
+  // def greet(name)
+  //   puts "howdy, #{name}!"
+  // end
+
+  // function greet(name) {
+  //   alert("howdy, " + name + "!");
+  // }
+
+  // greet("Raghu");
+
+
+  // Conditionals
+  // ============
+
+  // city = "Toledo"
+  // if city == "Chicago"
+  //   puts "You live in the best city ever!"
+  // else
+  //   puts "You live in a so-so city."
+  // end
+
+  // var city = "Toledo";
+  // if (city === "Chicago") {
+  //   alert("You live in the best city ever!");
+  // }
+  // else {
+  //   alert("You live in a so-so city.");
+  // }
+
+  // Beware: there's also a == (non-strict equivalence) operator. Look it up.
+
+
+  // Incrementing integers
+  // =====================
+
+  // i += 1
+
+  // i++
+
+
+  // For loops
+  // =========
+
+  // for i in 2...6
+  //   puts i * i
+  // end
+
+  // for (var i = 2; i < 6; i++) {
+  //   alert(i * i);
+  // }
```

Comment:
This is definitely the standard for loop in JS. However, in ES6 there is a closer equivalence when looping over Arrays, [`for...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of):

```js
let colors = ['red', 'green', 'blue'];
for (const color of colors){
    console.log(color);
}
```

_not to be confused with JS's `for...in`, which is designed to be used for looping over an objects properties_